src/neuroconv/tools/nwb_helpers/_models/_base_dataset_models.py

```diff
@@ -0,0 +1,180 @@
+            "\n"
+        )
+        if self.compression_method is not None:
+            string += f"\n  compression method : {self.compression_method}"
+        if self.compression_options is not None:
+            string += f"\n  compression options : {self.compression_options}"
+        if self.compression_method is not None or self.compression_options is not None:
+            string += "\n"
+        # TODO: would be cool to include estimate of ratio too (determined via stub file perhaps?)
+
+        return string
+
+    @root_validator
+    def validate_all_shapes(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        chunk_shape = values["chunk_shape"]
+        buffer_shape = values["buffer_shape"]
+        full_shape = values["dataset_info"].full_shape
+        location = values["dataset_info"].location  # For more identifiable error messages.
+
+        if len(chunk_shape) != len(buffer_shape):
+            raise ValueError(
+                f"{len(chunk_shape)=} does not match {len(buffer_shape)=} for dataset at location '{location}'!"
+            )
+        if len(buffer_shape) != len(full_shape):
+            raise ValueError(
+                f"{len(buffer_shape)=} does not match {len(full_shape)=} for dataset at location '{location}'!"
+            )
+
+        if any(chunk_axis <= 0 for chunk_axis in chunk_shape):
+            raise ValueError(
+                f"Some dimensions of the {chunk_shape=} are less than or equal to zero for dataset at "
+                f"location '{location}'!"
+            )
+        if any(buffer_axis <= 0 for buffer_axis in buffer_shape):
+            raise ValueError(
+                f"Some dimensions of the {buffer_shape=} are less than or equal to zero for dataset at "
+                f"location '{location}'!"
+            )
+
+        if any(chunk_axis > buffer_axis for chunk_axis, buffer_axis in zip(chunk_shape, buffer_shape)):
+            raise ValueError(
+                f"Some dimensions of the {chunk_shape=} exceed the {buffer_shape=} for dataset at "
+                f"location '{location}'!"
+            )
+        if any(buffer_axis > full_axis for buffer_axis, full_axis in zip(buffer_shape, full_shape)):
+            raise ValueError(
+                f"Some dimensions of the {buffer_shape=} exceed the {full_shape=} for dataset at "
+                f"location '{location}'!"
+            )
+
+        if any(buffer_axis % chunk_axis != 0 for chunk_axis, buffer_axis in zip(chunk_shape, buffer_shape)):
+            raise ValueError(
+                f"Some dimensions of the {chunk_shape=} do not evenly divide the {buffer_shape=} for dataset at "
+                f"location '{location}'!"
+            )
+
+        return values
+
+    @abstractmethod
+    def get_data_io_keyword_arguments(self):
+        """
+        Fetch the properly structured dictionary of input arguments to be passed directly into a H5DataIO or ZarrDataIO.
```

Comment:
```suggestion
        Fetch the properly structured dictionary of input arguments to be passed directly into a H5DataIO or ZarrDataIO. That is, you can wrap data using `wrapped_data=H5DataIO(**get_data_io_kwargs` or `wrapped_data=ZarrDataIO(**get_data_io_kwargs)`.
```

Would it be possible for this to aslo return the data so the above docstring is correct? We use this pattern in the tools to create all the keyword that are passed to create an object. I think we should honor this pattern here as well. 

Another option would be to have the kwargs for  `set_data_io` so we could do:

```python
        nwbfile_objects[object_id].set_data_io( **dataset_configuration.get_data_io_kwargs())
```
Instead of what we have now which is:
```python
        nwbfile_objects[object_id].set_data_io(dataset_name=dataset_name, data_io_class=data_io_class, **data_io_kwargs)
```

This is probably our main interest here as well.

Reply:
Answering to why the suggested docstring is incorrect:

* `H5DataIO` takes the data as well which is not returned by the current function. That is,  `wrapped_data=H5DataIO(**get_data_io_kwargs)` will not work.

I am trying to keep consistency with the pattern that we use in the tools (spikeinterface and roiextractors tools in neuroconv I mean):

Examples:
https://github.com/catalystneuro/neuroconv/blob/9efdefa5963ecca4ea6cd8ff392ec0b6be903702/src/neuroconv/tools/spikeinterface/spikeinterface.py#L679-L680 

https://github.com/catalystneuro/neuroconv/blob/9efdefa5963ecca4ea6cd8ff392ec0b6be903702/src/neuroconv/tools/roiextractors/roiextractors.py#L388-L393

But I actually think that trying to avoid dynamic passing is a worthwhile goal. I am willing to consider other options. I have some ideas but maybe we can discuss that in another PR down the chain?

Reply:
Ah, but these are not intended to be the `kwargs` that get passed into the function `.set_data_io`, but instead they fill the role of the third argument of the `.set_data_io` method

Note that if `wrapped_data=H5DataIO(**get_data_io_kwargs)` works under your model, that might only be because the `data_io_class` keyword argument would be ignored/skipped because it's not in the signature

Hence I prefer mine for readability as well; 'set my data IO on the dataset with `dataset_name` using the `data_io_class` of my choosing, which takes and uses additional keyword arguments structured by `.get_data_io_kwargs()`'